openupgradelib/openupgrade.py

```diff
@@ -196,21 +199,27 @@ def allow_pgcodes(cr, *codes):
         that are allowed. Codes can have either 2 characters (indicating an
         error class) or 5 (indicating a concrete error). Any other errors
         will be raised.
+    :param query:
+        When passed, the query that causes the exception will be logged as
+        well. Has to be passed as a keyword argument.
     """
     try:
-        from psycopg2 import errorcodes, ProgrammingError
+        from psycopg2 import errorcodes, ProgrammingError, IntegrityError
     except ImportError:
-        from psycopg2cffi import errorcodes, ProgrammingError
+        from psycopg2cffi import errorcodes, ProgrammingError, IntegrityError
 
     try:
         with cr.savepoint():
-            yield
-    except ProgrammingError as error:
+            with core.tools.mute_logger('odoo.sql_db'):
+                yield
+    except (ProgrammingError, IntegrityError) as error:
+        if kwargs.get('query'):
+            logger.info(kwargs['query'])
         msg = "Code: {code}. Class: {class_}. Error: {error}.".format(
             code=error.pgcode,
             class_=errorcodes.lookup(error.pgcode[:2]),
             error=errorcodes.lookup(error.pgcode))
-        if error.pgcode not in codes and error.pgcode[:2] in codes:
+        if error.pgcode in codes or error.pgcode[:2] in codes:
```

Comment:
Fixing what looks like an obvious bug after a refactoring. @yajo?

Reply:
Yes, so it seems, thanks!